File: Shared/TransitionFilterCriteria.vb
'************************************************************************************
' ST-Sim: A .NET class library for developing state-and-transition simulation models.
'
' Copyright © 2007-2017 Apex Resource Management Solution Ltd. (ApexRMS). All rights reserved.
'
'************************************************************************************

''' <summary>
''' Transition filter criteria class
''' </summary>
''' <remarks></remarks>
Class TransitionFilterCriteria

    Public IncludeDeterministic As Boolean = True
    Public IncludeProbabilistic As Boolean = True
    Public TransitionGroups As New Dictionary(Of Integer, Boolean)

End Class

File: Shared/TransitionFilterCriteria.cs
// ST-Sim: A SyncroSim Module for the ST-Sim State-and-Transition Model.
// Copyright © 2007-2018 Apex Resource Management Solution Ltd. (ApexRMS). All rights reserved.

using System.Collections.Generic;

namespace SyncroSim.STSim
{
    internal class TransitionFilterCriteria
    {
        public bool IncludeDeterministic = true;
        public bool IncludeProbabilistic = true;
        public Dictionary<int, bool> TransitionGroups = new Dictionary<int, bool>();
    }
}
